packages/core/features/src/signAndSendTransaction.ts

```diff
@@ -63,4 +63,7 @@ export type SolanaSignAndSendTransactionOptions = SolanaSignTransactionOptions &
 
     /** Maximum number of times for the RPC node to retry sending the transaction to the leader. */
     readonly maxRetries?: number;
+
+    /** Wait till specified commitment before sending the next transaction. If commitment not specified, wait till "confirmed" commitment*/
+    readonly waitForCommitmentToSendNextTransaction?: boolean;
```

Comment:
Hmm, if `commitment` is not specified, I'm not sure we can rely upon the `wait` semantics... I'll think about this a bit more.

Reply:
if `commitment` is not specified, but `waitForCommitmentToSendNextTransaction` is, we could use `confirmed` as the default commitment to be reached before sending the next transaction. For most cases, confirmed provides a sufficiently good promise of transaction submission.

Reply:
I'm no Solana expert, so I don't know the answer to this question: are there dApps out there using `signAndSendTransaction` passing an array of transactions (effectively `signAndSendAllTransactions`) and including a `commitment` level without expecting sequential sending?

In other words: instead of adding a new option, could we extend the behavior of the `commitment` field to be:

"confirm the transaction after sending it and before sending the next transaction"

?

Reply:
@olistic I'm actually thinking of deprecating `commitment`. I don't actually think wallets use it, and the only useful value for it is `confirmed` which is what wallets use by default anyway when confirming.

Right now no apps that I know of call `signAndSendTransaction` with multiple transactions because Wallet Adapter doesn't support it. After we make this change, we would add a `signAndSendAllTransactions` method to Wallet Adapter.